sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/Catalogs.scala

```diff
@@ -45,7 +45,11 @@ private[sql] object Catalogs {
   @throws[SparkException]
   def load(name: String, conf: SQLConf): CatalogPlugin = {
     val pluginClassName = try {
-      conf.getConfString("spark.sql.catalog." + name)
+      val _pluginClassName = conf.getConfString(s"spark.sql.catalog.$name")
+      if (name.contains(".")) {
```

Comment:
shall we validate it before getting conf?

Reply:
Then we need a specific exception class for that case like `InvalidCatalogNameException`, and catch it as well as   `CatalogNotFoundException` in`CatalogManager` and `LookupCatalog`, otherwise may casue failure in analyze phase
```
- run sql directly on files *** FAILED *** (369 milliseconds)
[info]   org.apache.spark.SparkException: Invalid catalog name: org.apache.spark.sql.json
[info]   at org.apache.spark.sql.errors.QueryExecutionErrors$.invalidCatalogNameError(QueryExecutionErrors.scala:1538)
[info]   at org.apache.spark.sql.connector.catalog.Catalogs$.load(Catalogs.scala:47)
[info]   at org.apache.spark.sql.connector.catalog.CatalogManager.$anonfun$catalog$1(CatalogManager.scala:54)
[info]   at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:86)
[info]   at org.apache.spark.sql.connector.catalog.CatalogManager.catalog(CatalogManager.scala:54)
[info]   at org.apache.spark.sql.connector.catalog.LookupCatalog$CatalogAndIdentifier$.unapply(LookupCatalog.scala:122)
[info]   at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.$anonfun$lookupRelation$1(Analyzer.scala:1199)
```
Do you think it's a right direction?

Reply:
I see. Let's get conf first then, and also add some comments to explain the rationale.

Reply:
Got it, updated.